src/decompose_mesh/rules.mk

```diff
@@ -106,7 +130,10 @@ endif
 
 $O/decompose_mesh.dec.o: $O/part_decompose_mesh.dec.o $O/fault_scotch.dec.o ${SCOTCH_DIR}/include/scotchf.h $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
 $O/program_decompose_mesh.dec.o: $O/decompose_mesh.dec.o $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
-
+$O/program_decompose_mesh_mpi.mpidec.o: $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
```

Comment:
Thanks!
Fixed.

Elliott when you have time could you please check our fix?
(we are not very familiar with what to put in rules.mk)
Thanks.


On 02/10/2016 07:26 PM, Elliott Sales de Andrade wrote:
> In src/decompose_mesh/rules.mk
> <https://github.com/geodynamics/specfem3d/pull/694#discussion_r52500943>:
>
>> @@ -106,7 +130,10 @@ endif
>>
>>  $O/decompose_mesh.dec.o: $O/part_decompose_mesh.dec.o $O/fault_scotch.dec.o ${SCOTCH_DIR}/include/scotchf.h $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
>>  $O/program_decompose_mesh.dec.o: $O/decompose_mesh.dec.o $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
>> -
>> +$O/program_decompose_mesh_mpi.mpidec.o: $O/shared_par.shared_module.o $(COND_MPI_OBJECTS)
>
> There are a few other dependencies missing, I think. From my look through:
>
>   * decompose_mesh/module_mesh.f90
>       o fault_scotch
>   * decompose_mesh/module_partition.f90
>       o module_qsort
>   * decompose_mesh/program_decompose_mesh_mpi.f90
>       o module_mesh
>       o module_database
>       o module_partition
>
> Also, it is not necessary to specify |$O/shared_par.shared_module.o|
> here; it is already a dependency from the |$O/%.dec.o| rule below.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/geodynamics/specfem3d/pull/694/files#r52500943>.
>

-- 
Dimitri Komatitsch
CNRS Research Director (DR CNRS), Laboratory of Mechanics and Acoustics,
UPR 7051, Marseille, France    http://komatitsch.free.fr